code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -112,123 +75,132 @@ This saves us from having to call add_fingerprint() any time something is put in
-		if(!(head && (head.item_flags & ITEM_FLAG_AIRTIGHT)))
-			set_internals(null)
-		update_inv_wear_mask()
-	else if (W == wear_id)
-		wear_id = null
-		update_inv_wear_id()
-	else if (W == r_store)
-		r_store = null
-		update_inv_pockets()
-	else if (W == l_store)
-		l_store = null
-		update_inv_pockets()
-	else if (W == s_store)
-		s_store = null
-		update_inv_s_store()
-	else if (W == back)
-		back = null
-		update_inv_back()
-	else if (W == handcuffed)
-		handcuffed = null
-		if(buckled && buckled.buckle_require_restraints)
-			buckled.unbuckle_mob()
-		update_inv_handcuffed()
-	else if (W == r_hand)
-		r_hand = null
-		if(l_hand)
-			l_hand.update_twohanding()
-			update_inv_l_hand()
-		update_inv_r_hand()
-	else if (W == l_hand)
-		l_hand = null
-		if(r_hand)
-			r_hand.update_twohanding()
-			update_inv_l_hand()
-		update_inv_l_hand()
-	else
-		return 0
-
-	update_action_buttons()
-	return 1
-
-
+			if(r_ear == W) //check for items that get equipped to both ear slots
+				r_ear = null
+			update_inv_ears()
+		else if (W == r_ear)
+			r_ear = null
+			if(l_ear == W)
+				l_ear = null
+			update_inv_ears()
+		else if (W == shoes)
+			shoes = null
+			update_inv_shoes()
+		else if (W == belt)
+			belt = null
+			update_inv_belt()
+		else if (W == wear_mask)
+			wear_mask = null
+			if(istype(W, /obj/item))
+				var/obj/item/I = W
+				if(I.flags_inv & (BLOCKHAIR|BLOCKHEADHAIR))
+					update_hair(0)	//rebuild hair
+					update_inv_ears(0)
+			var/obj/item/clothing/mask/head = src.get_equipped_item(slot_head)
+			if(!(head && (head.item_flags & ITEM_FLAG_AIRTIGHT)))
+				set_internals(null)
+			update_inv_wear_mask()
+		else if (W == wear_id)
+			wear_id = null
+			update_inv_wear_id()
+		else if (W == r_store)
+			r_store = null
+			update_inv_pockets()
+		else if (W == l_store)
+			l_store = null
+			update_inv_pockets()
+		else if (W == s_store)
+			s_store = null
+			update_inv_s_store()
+		else if (W == back)
+			back = null
+			update_inv_back()
+		else if (W == handcuffed)
+			handcuffed = null
+			if(buckled && buckled.buckle_require_restraints)
+				buckled.unbuckle_mob()
+			update_inv_handcuffed()
+		else
+			. = FALSE
+	if(.)
+		update_action_buttons()
 
 //This is an UNSAFE proc. Use mob_can_equip() before calling this one! Or rather use equip_to_slot_if_possible() or advanced_equip_to_slot_if_possible()
 //set redraw_mob to 0 if you don't wish the hud to be updated - if you're doing it manually in your own proc.
 /mob/living/carbon/human/equip_to_slot(obj/item/W, slot, redraw_mob = 1)
 
-	if(!slot) return
-	if(!istype(W)) return
-	if(!has_organ_for_slot(slot)) return
-	if(!species || !species.hud || !(slot in species.hud.equip_slots)) return
-	W.forceMove(src)
+	. = ..()
+	if(!. || !has_organ_for_slot(slot))
+		return
 
+	// TODO: unify this block with below when inventory
+	// is rewritten to remove boilerplate horseshit.
+	u_equip(W)
 	var/obj/item/old_item = get_equipped_item(slot)
+	if(!isnum(slot))
```

Comment:
I'm pretty lost here (apparently can be num or string, giving different function?). I appreciate that there's a todo above, but may help to give some comments explaining roughly what's going on here and what the expectations are. What do the return values mean? What is `slot` here? etc.